samples/cellular/http_update/modem_delta_update/src/main.c

```diff
@@ -307,9 +309,9 @@ static int update_download(void)
 	}
 
 	/* Functions for getting the host and file */
-	err = fota_download_start(CONFIG_DOWNLOAD_HOST, file, SEC_TAG, 0, 0);
+	err = fota_download_any(CONFIG_DOWNLOAD_HOST, file, &sec_tag, sec_tag_count, 0, 0);
```

Comment:
Actually.. As we are touching these, why don't we change it to use `DFU_TARGET_IMAGE_TYPE_MODEM_DELTA`

After all, this is DELTA update demo.

So it is kind of wrong to accept any kind of image.